authenticate/state.go

```diff
@@ -72,16 +72,16 @@ func newAuthenticateStateFromConfig(cfg *config.Config) (*authenticateState, err
 	state.redirectURL, _ = urlutil.DeepCopy(authenticateURL)
 	state.redirectURL.Path = cfg.Options.AuthenticateCallbackPath
 
+	// shared cipher to encrypt data before passing data between services
+	state.sharedKey, _ = base64.StdEncoding.DecodeString(cfg.Options.SharedKey)
+	state.sharedCipher, _ = cryptutil.NewAEADCipher(state.sharedKey)
```

Comment:
let's not ignore errors

Reply:
I guess we've been through this a few times, but errors are checked upstream. I didn't change this behavior.

I will update the code but it will hurt readability.

Reply:
I looked how this function is called, and there's no other upstream that checks and validates these parameters. This function is parsing and preparing the runtime object, and it has to return all errors. 

btw following that logic, you should also drop the `cfg.Options.Validate()` call at the very top of this function, as that is an example of a call that was supposed to be called upstream.

Reply:
Shared key is checked with the call to Validate in config.